libvast/src/system/evaluator.cpp

```diff
@@ -153,7 +153,17 @@ evaluator(evaluator_actor::stateful_pointer<evaluator_state> self,
   self->state.expr = std::move(expr);
   self->state.eval = std::move(eval);
   self->state.store = std::move(store);
+  self->send(self->state.store, atom::exporter_v,
+             caf::actor_cast<caf::actor>(self));
   return {
+    [self](table_slice slice) {
+      self->send(self->state.client, std::move(slice));
```

Comment:
Should this `return self->delegate` rather than `self->send`?

Reply:
I'm not sure how delegate fits here?

Reply:
The handler of the archive/store returns a `caf::result<void>`. You don't do error handling here, but the actor that sends the slices to the evaluator might want to do that. By delegating you forward errors that you receive.